addons/website_sale/static/src/website_builder/product_page_option.js

```diff
@@ -8,14 +8,15 @@ export class ProductPageOption extends BaseOptionComponent {
     setup() {
         super.setup();
         this.domState = useDomState((el) => {
+            const productDetailEl = el.querySelector("#product_detail");
             const productDetailMainEl = el.querySelector("#product_detail_main");
             const productPageCarouselEl = el.querySelector("#o-carousel-product");
             const productPageGridEl = el.querySelector("#o-grid-product");
-            const hasImages = productDetailMainEl.dataset.image_width !== "none";
-            const isFullImage = productDetailMainEl.dataset.image_width === "100_pc";
+            const hasImages = !productDetailEl.classList.contains("o_wsale_image_width_none");
```

Comment:
nitpicking: Why can't we rely on the dataset anymore? I'd like to keep using classes for styling only, use names for selectors, and datasets for properties.

Reply:
I see, and that would make sense indeed.
The issue is that during edition we can’t rely on dataset, because the default `PreviewableWebsiteConfigAction` editor action toggles classes and not data attributes. Data attributes are only applied to the DOM **after** saving, so if we check for them during edition, the `domState` won’t reflect what the user actually sees on screen.

That said, I haven’t changed how data attributes are eventually set in the DOM, the information is still there if needed. It’s just not reliable in this specific context 😕

Reply:
Ok, I'll check with website if we can improve that but let's keep it at the moment.